galley/pkg/fs/fssource.go

```diff
@@ -303,6 +304,9 @@ func (s *fsSource) Start() (chan resource.Event, error) {
 
 // New returns a File System implementation of runtime.Source.
 func New(root string, config *converter.Config) (runtime.Source, error) {
+	if config.ConvertK8SService {
```

Comment:
Is this needed? While having K8s services fed through file system doesn't make much sense, it still doesn't hurt in this context. 

In general, it would make more sense to pass the metadata as a parameter to the K8s & FS source methods, rather than try to calculate in each one of them. pkg/server is probably the right location to do the calculation before instantiating the appropriate source.

Reply:
Do you mean to pass []kube.ResourceSpec in converter.Config or as a separate argument? The latter might work, but the former would create an import cycle between kube and kube/converter